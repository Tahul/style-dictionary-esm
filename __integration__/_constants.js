import { fileURLToPath } from 'url'
import { dirname, join, resolve } from 'pathe'

const _dir = resolve(dirname(fileURLToPath(import.meta.url)))
export const resolveBuildPath = _path => join(_dir, '__integration__/build/')

export const buildPath = resolveBuildPath()

export const clearConsoleOutput = (str) => {
  const arr = str.split('\n')
    // Remove ANSI stuff from the console output so we get human-readable strings
    // https://github.com/chalk/ansi-regex/blob/main/index.js#L3
    .map(s => s.replace(/[\u001B\u009B][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g, '').trim())
  return arr.join('\n')
}
