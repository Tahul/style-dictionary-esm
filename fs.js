// Allow to be overridden by setter, set default to memfs for browser env, node:fs for node env
export let fs = undefined;

// since ES modules exports are read-only, use a setter
export const setFs = (_fs) => {
  fs = _fs;
};
