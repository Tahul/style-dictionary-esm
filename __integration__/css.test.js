/*
 * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import fs from 'fs-extra'
import { join } from 'pathe'
import StyleDictionary from '../src/index'
import { buildPath } from './_constants'

describe('integration', () => {
  describe('css', () => {
    StyleDictionary.extend({
      source: ['__integration__/tokens/**/*.json?(c)'],
      // Testing proper string interpolation with multiple references here.
      // This is a CSS/web-specific thing so only including them in this
      // integration test.
      properties: {
        breakpoint: {
          xs: { value: '304px' },
          sm: { value: '768px' },
          md: { value: 'calc({breakpoint.xs.value} / {breakpoint.sm.value})' },
        },
      },
      platforms: {
        css: {
          transformGroup: 'css',
          buildPath,
          files: [{
            destination: 'variables.css',
            format: 'css/variables',
          }, {
            destination: 'variablesWithReferences.css',
            format: 'css/variables',
            options: {
              outputReferences: true,
              outputReferenceFallbacks: true,
            },
          }, {
            destination: 'variablesWithSelector.css',
            format: 'css/variables',
            options: {
              selector: '.test',
            },
          }],
        },
      },
    }).buildAllPlatforms()

    describe('css/variables', () => {
      it('should match snapshot', () => {
        const output = fs.readFileSync(join(buildPath, 'variables.css'), { encoding: 'UTF-8' })
        expect(output).toMatchSnapshot()
      })

      describe('with references', () => {
        it('should match snapshot', () => {
          const output = fs.readFileSync(join(buildPath, 'variablesWithReferences.css'), { encoding: 'UTF-8' })
          expect(output).toMatchSnapshot()
        })
      })

      describe('with selector', () => {
        it('should match snapshot', () => {
          const output = fs.readFileSync(join(buildPath, 'variablesWithSelector.css'), { encoding: 'UTF-8' })
          expect(output).toMatchSnapshot()
        })
      })

      // TODO: add css validator
    })
  })
})

afterAll(() => {
  fs.emptyDirSync(buildPath)
})
