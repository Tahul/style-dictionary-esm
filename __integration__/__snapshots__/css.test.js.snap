// Vitest Snapshot v1

exports[`integration > css > css/variables > should match snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Thu, 01 Sep 2022 01:39:44 GMT
 */

:root {
  --breakpoint-xs: 304px;
  --breakpoint-sm: 768px;
  --breakpoint-md: calc(304px / 768px);
}
"
`;

exports[`integration > css > css/variables > with references > should match snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Thu, 01 Sep 2022 01:39:44 GMT
 */

:root {
  --breakpoint-sm: 768px;
  --breakpoint-xs: 304px;
  --breakpoint-md: calc(var(--breakpoint-xs) / var(--breakpoint-sm));
}
"
`;

exports[`integration > css > css/variables > with selector > should match snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Thu, 01 Sep 2022 01:39:44 GMT
 */

.test {
  --breakpoint-xs: 304px;
  --breakpoint-sm: 768px;
  --breakpoint-md: calc(304px / 768px);
}
"
`;
